### ISSUE TITLE
OnClick Remove or Add a Draggable Element

### ISSUE THREAD

**Issue by hariria** [atlassian/react-beautiful-dnd#1518]
Hi,

I'm trying to create two functions. 
1. A function that lets me delete a draggable element
2. A function that lets me add a draggable element

In the codesandbox below, I have created a trashcan icon that is supposed to delete elements onclick and set the state. It sets the state correctly, but for some reason it's not updated in the draggable list. I think that if I can figure out how to delete an element properly, I'll be able to figure out how to add an element. I've been stuck on this for some time now, if you could help me out I'd really appreciate it!

https://codesandbox.io/s/hopeful-moore-zsrmj?fontsize=14

**Comment by tberghuis** [comment]
Use he spread operator in setItems call line 123, https://codesandbox.io/s/serene-ganguly-1o359
Article about useState hook, https://blog.logrocket.com/a-guide-to-usestate-in-react-ecb9952e406c/
tldr useState hook does a shallow comparison

**Comment by qas** [comment]
This is a bug, so maybe it should be labeled as one.

**Comment by jamesdamild** [comment]
hi i am also stuck at a similar task, any help, please?

**Comment by danieldelcore** [comment]
Closing this one due to inactivity. Please feel free to reopen if need be.
